tcti/tcti.h

```diff
@@ -69,12 +69,42 @@ typedef struct {
     UINT32 size;
     UINT32 code;
 } tpm_header_t;
-
-enum tctiStates { TCTI_STAGE_INITIALIZE, TCTI_STAGE_SEND_COMMAND, TCTI_STAGE_RECEIVE_RESPONSE };
+/*
+ * The elements in this enumeration represent the possible states that the
+ * TCTI can be in. The state machine is as follows:
+ * An instantiated TCTI context begins in the TRANSMIT state:
+ *   TRANSMIT:
+ *     transmit:    success transitions the state machine to RECEIVE
+ *                  failure leaves the state unchanged
+ *     receieve:    produces TSS2_TCTI_RC_BAD_SEQUENCE
+ *     finalize:    transitions state machine to FINAL state
+ *     cancel:      produces TSS2_TCTI_RC_BAD_SEQUENCE
+ *     setLocality: success or failure leaves state unchanged
+ *   RECEIVE:
+ *     transmit:    produces TSS2_TCTI_RC_BAD_SEQUENCE
+ *     receive:     success transitions the state machine to TRANSMIT
+ *                  failure with the following RCs leave the state unchanged:
+ *                    TRY_AGAIN, INSUFFICIENT_BUFFER, BAD_CONTEXT,
+ *                    BAD_REFERENCE, BAD_VALUE, BAD_SEQUENCE
+ *                  all other failures transition state machine to
+ *                    TRANSMIT (not recoverable)
+ *     finalize:    transitions state machine to FINAL state
+ *     cancel:      success transitions state machine to READY_TRANSMIT
```

Comment:
Where is the READY_TRANSMIT defined? or is it just STATE_TRANSMIT?

Reply:
That's an old name for the `TRANSMIT` state (aka `TCTI_STATE_TRANSMIT`). Will fix this up.